src/anomaly/automl_regressors.rs

```diff
@@ -23,18 +26,18 @@ fn metaml_dataset_to_automl(data: MetaMLDataset) -> Dataset<f32, f32> {
         "Radius (exponential moving average)".to_string(),
         "Local fractal dimension (exponential moving average)".to_string(),
     ];
-    let target_names = vec!["Score".to_string()];
+    let target_names = vec!["Expected AUC".to_string()];
     let description = "A dataset for training a meta-ml model to predict ROC AUC based on six inputs".to_string();
 
     // Construct and return a dataset
     Dataset {
-        num_samples: data.features.nrows(),
+        num_samples: data.features.len(),
         num_features: 6,
         feature_names,
         target_names,
         description,
-        data: data.features.into_raw_vec(),
-        target: data.targets.into_raw_vec(),
+        data: data.features.into_iter().flatten().collect(),
```

Comment:
This is a data copy, right? If so, we do not want data copies.

Reply:
**TL;DR** This isn't doing any more data copying than before (just more explicitly), and optimizing to avoid any data copying would be a pretty big hit to code coupling (but I'll do it if you want me to anyway of course :))

At a high level, no, this is not a data copy. `.into_iter()`, `.flatten()`, and `.collect()` all work with owned values here.

At a more low level, this *might* be doing some data copying in `.collect()` due to the `.flatten()` (if the underlying data wasn't previously contiguous in memory). If this were the case though, that same copying would have been occurring elsewhere - just less explicitly since it's not in our code, but in another function we call somewhere. This copying is necessary to exist *somewhere* because we're working with two different APIs (`automl` and `ndarray-npy`) which give and expect the same data in different formats. This change does the data format transformation when creating the MetaMLDataset instead of when using it.

If this is a performance issue, we could modify our code so that MetaMLDataset's internal representation is tied to the output format from `ndarray-npy`, and modify `automl`'s code to expect that same format. This will eliminate the need for copying, but will significantly increase the amount of coupling and cost of changes/expansion in the future... in my opinion, that tradeoff isn't worth it unless we find through benchmarking that this is actually a performance issue. If you feel differently, absolutely let me know and I'll refactor the code with performance in mind (although that will take some time)

Reply:
This is the correct answer: profile and benchmark, and see if there's actually any performance bottleneck worth dealing with. I somehow doubt it.